test/integration/directory_artifact_test.js

```diff
@@ -79,4 +83,81 @@ suite('Directory artifact', () => {
       'Size of uploaded contents of test.html does not match original.'
     );
   });
+
+  test('retry directory upload', async () => {
+    let ARTIFACT_COUNT = 3;
+
+    let worker = new TestWorker(DockerWorker);
+    await worker.launch();
+
+    var count = 0;
+
+    let rejectConnection = function() {
+      // Delay iptables rejection to catch
+      // in the middle of the upload
+      setTimeout(function() {
+        iptables.reject({
```

Comment:
That's the only way we can get artifacts retry integration tests so far, which is very important, IMO. Although it has its drawbacks, testing I think the using iptables is still worth.